solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -393,14 +393,22 @@ private SolrConfig(
   }
 
   private IndexSchemaFactory.VersionedConfig readXml(SolrResourceLoader loader, String name) {
+    InputStream in = null;
     try {
-      ResourceProvider rp = new ResourceProvider(loader, name);
+      in = loader.openResource(name);
+      ResourceProvider rp = new ResourceProvider(in);
       XmlConfigFile xml = new XmlConfigFile(loader, rp, name, null, "/config/", null);
       return new IndexSchemaFactory.VersionedConfig(
           rp.zkVersion,
           new DataConfigNode(new DOMConfigNode(xml.getDocument().getDocumentElement())));
     } catch (IOException e) {
       throw new SolrException(ErrorCode.SERVER_ERROR, e);
+    } finally {
+      // according to spec, XML parser should close InputStream when parsing is complete.
+      // But in the event that this doesn't happen (either because an exception is
+      // thrown or because of an error in parser implementation, here we ensure that it
+      // is closed.
```

Comment:
BTW I appreciate code comments but I don't think you need to be so wordy to justify closing ourselves.  IMO this can be summarized to one line:
> XML Parser *should* close but in exceptional cases might not; so let's be safe

Reply:
addressed in 9507e1ae3be06daaf7400fa41e77a75d3fb55981